python/ray/train/torch/torch_predictor.py

```diff
@@ -21,17 +21,32 @@ class TorchPredictor(Predictor):
         model: The torch module to use for predictions.
         preprocessor: A preprocessor used to transform data batches prior
             to prediction.
+        use_gpu: If prediction is done on GPU where model will be moved to
+            GPU device upon instantiation.
     """
 
     def __init__(
-        self, model: torch.nn.Module, preprocessor: Optional["Preprocessor"] = None
+        self,
+        model: torch.nn.Module,
+        preprocessor: Optional["Preprocessor"] = None,
+        use_gpu: bool = False,
     ):
         self.model = model
+        self.model.eval()
         self.preprocessor = preprocessor
 
+        # TODO (jiaodong): #26249 Use multiple GPU devices with sharded input
+        self.use_gpu = use_gpu
+        if use_gpu:
+            # Ensure input tensor and model live on GPU for GPU inference
+            self.model.to(torch.device("cuda"))
```

Comment:
model init and config is moved to constructor rather than upon every predict() call.